src/nvim/os/stdpaths.c

```diff
@@ -69,13 +69,19 @@ const char *get_appname(void)
   return env_val;
 }
 
-/// Ensure that APPNAME is valid. In particular, it cannot contain directory separators.
+// Ensure that APPNAME is valid. In particular, it cannot contain .. surrounded
+// by path separators, the start of the string, or the end of the string.
```

Comment:
No reason to allow ".." at all. That simplifies both the documentation and the logic.

```suggestion
// Ensure that APPNAME is valid. Must be a name or relative path.
```

Reply:
What if a user has a directory name like `nvim..configs`? It's weird, I don't like it, but it's a valid dir name